wandb/sdk/launch/agent/agent.py

```diff
@@ -588,7 +599,20 @@ def _thread_run_job(
             return
         with self._jobs_lock:
             job_tracker.run = run
+        start_time = time.time()
         while self._jobs_event.is_set():
+            # If run has failed to start before timeout, kill it
+            if job_tracker.run.get_status().state == "starting":
```

Comment:
this function would look better if this startup math was in a util function

Reply:
I tried to clean it up a bit without moving it to another function, it feels right to me to have the check here instead of passing the start time and the submitted_run to another function